src/TableFunctions/TableFunctionFuzzJSON.cpp

```diff
@@ -0,0 +1,108 @@
+{
+    ASTs & args_func = ast_function->children;
+
+    if (args_func.size() != 1)
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Table function '{}' must have arguments.", getName());
+
+    ASTs & args = args_func[0]->children;
+    if (args.empty())
+    {
+        source = "{}";
+        return;
+    }
+
+    static constexpr size_t max_args = 3;
+    if (args.size() > max_args)
+        throw Exception(
+            ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Table function '{}' requires at most three arguments: [source, random_seed, max_output_length].", getName());
+
+    // All the arguments must be literals.
+    for (const auto & arg : args)
+    {
+        if (!arg->as<const ASTLiteral>())
+        {
+            throw Exception(
+                ErrorCodes::BAD_ARGUMENTS,
+                "All arguments of the table function '{}' must be literals. "
+                "Got '{}' instead",
+                getName(),
+                arg->formatForErrorMessage());
+        }
+    }
+
+    size_t arg_index = 0;
+
+    source = checkAndGetLiteralArgument<String>(args[arg_index], "source");
+    ++arg_index;
+
+    if (args.size() >= arg_index + 1)
+    {
+        const auto & literal = args[arg_index]->as<const ASTLiteral &>();
+        if (!literal.value.isNull())
+            random_seed = checkAndGetLiteralArgument<UInt64>(literal, "random_seed");
+        ++arg_index;
+    }
+
+    if (args.size() == arg_index + 1)
+    {
+        max_output_length = checkAndGetLiteralArgument<UInt64>(args[arg_index], "max_output_length");
+        if (max_output_length < 2)
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "The value of 'max_output_length' argument of the table function '{}' must be greater or equal than 2", getName());
+    }
+}
+
+ColumnsDescription TableFunctionFuzzJSON::getActualTableStructure(ContextPtr /*context*/, bool /*is_insert_query*/) const
+{
+    return ColumnsDescription{{"json", std::make_shared<DataTypeString>()}};
+}
+
+StoragePtr TableFunctionFuzzJSON::executeImpl(
+    const ASTPtr & /*ast_function*/,
+    ContextPtr context,
+    const std::string & table_name,
+    ColumnsDescription /*cached_columns*/,
+    bool is_insert_query) const
+{
+    auto config = StorageFuzzJSON::Config{};
+    if (random_seed)
+        config.random_seed = *random_seed;
+    if (max_output_length)
+        config.max_output_length = *max_output_length;
+
+    ColumnsDescription columns = getActualTableStructure(context, is_insert_query);
+    auto res = std::make_shared<StorageFuzzJSON>(
+        StorageID(getDatabaseName(), table_name),
+        columns,
+        /* comment */ String{},
+        config,
+        source);
+    res->startup();
+    return res;
+}
+
+void registerTableFunctionFuzzJSON(TableFunctionFactory & factory)
+{
+    factory.registerFunction<TableFunctionFuzzJSON>({.documentation = {}, .allow_readonly = true});
```

Comment:
You may want to add documentation here instead of adding function to `02414_all_new_table_functions_must_be_documented`. Just copy the text you have written in `docs/`.

Examples:

https://github.com/ClickHouse/ClickHouse/blob/be88873c3e857f1219e78341b8155ad6c99ac9cf/src/TableFunctions/TableFunctionExplain.cpp#L179-L187

https://github.com/ClickHouse/ClickHouse/blob/be88873c3e857f1219e78341b8155ad6c99ac9cf/src/TableFunctions/TableFunctionFormat.cpp#L132-L202